client/signup/steps/domains/index.jsx

```diff
@@ -317,30 +317,19 @@ class DomainsStep extends React.Component {
 	};
 
 	shouldIncludeDotBlogSubdomain() {
-		const { flowName, isDomainOnly, siteGoals, signupDependencies } = this.props;
-		const siteGoalsArray = siteGoals ? siteGoals.split( ',' ) : [];
-
-		// 'subdomain' flow coming from .blog landing pages
-		if ( flowName === 'subdomain' ) {
-			return true;
-		}
-
-		// 'blog' flow, starting with blog themes
-		if ( flowName === 'blog' ) {
-			return true;
-		}
-
-		// No .blog subdomains for domain only sites
-		if ( isDomainOnly ) {
+		// Disable for domain only sites
+		if ( this.props.isDomainOnly ) {
 			return false;
 		}
 
-		// If we detect a 'blog' site type from Signup data
+		// Enable for 'subdomain' flow, originating from .blog landing pages
+		// Enable if the query includes ".blog"
+		// Enable if the user chooses `blog` as their site type
+		const lastQuery = get( this.props.step, 'domainForm.lastQuery' );
 		return (
-			// All flows where 'about' step is before 'domains' step, user picked only 'share' on the `about` step
-			( siteGoalsArray.length === 1 && siteGoalsArray.indexOf( 'share' ) !== -1 ) ||
-			// Users choose `Blog` as their site type
-			'blog' === get( signupDependencies, 'siteType' )
+			this.props.flowName === 'subdomain' ||
```

Comment:
As mentioned [above](https://github.com/Automattic/wp-calypso/pull/35347/files#r316962784), we probably don't need this flow name check.